### PR TITLE
Feature/31/refactor web3 service

### PR DESCRIPTION
**Pull request by KaliszukTomasz**
Feature/23/-user-service
Feature/31/refactor web3 service
Refresh after logout

### CODE REVIEW

File: src/app/admin/model/marketplace.ts [hunk 1]
```diff
@@ -1,15 +1,15 @@
 import { User } from "../model/user";
-import { MarketplaceStorage } from "../../model/marketplace";
+import { MarketplaceFromStorage } from "../../model/marketplace";
 
 export class Marketplace {
```
Comment (lukaszkinder) [inline src/app/admin/model/marketplace.ts]: this class should not have this copy constructor, Marketplace should not know anything about MarketplaceFromStorage. leave here only all args constructor
Create separate Factory class next to mocked services

File: src/app/admin/service/admin-marketplace-service-mock.service.ts [hunk 2]
```diff
@@ -32,44 +26,35 @@ export class AdminMarketplaceServiceMockService implements AdminMarketplaceServi
...
 
   createMarketplace(marketplaceName: string): Observable<MarketplaceMin> {
-   
-    var newMarketplaceToStorage: MarketplaceStorage = new MarketplaceStorage();
-    newMarketplaceToStorage.address = this.addressGenerator;
-    newMarketplaceToStorage.name = marketplaceName;
-    newMarketplaceToStorage.registeredUsers = [];
-    newMarketplaceToStorage.auctions = [];
 
+    var addressGenerator = "0x" + Math.random().toString(20).substring(2, 20);
```
Comment (lukaszkinder) [inline src/app/admin/service/admin-marketplace-service-mock.service.ts]: extract method for this generator

File: src/app/admin/service/admin-marketplace-service-mock.service.ts [hunk 3]
```diff
@@ -3,26 +3,20 @@ import { AdminMarketplaceService } from './admin-marketplace-service';
...
+import { MarketplaceFromStorage } from '../../model/marketplace';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AdminMarketplaceServiceMockService implements AdminMarketplaceService {
 
-
   marketplaces: Marketplace[] = [];
   marketplacesMin: MarketplaceMin[] = [];
```
Comment (lukaszkinder) [inline src/app/admin/service/admin-marketplace-service-mock.service.ts]: no need to have it as attribute

File: src/app/model/marketplace.ts [hunk 4]
```diff
@@ -1,16 +1,23 @@
...
-export class MarketplaceStorage {
+export class MarketplaceFromStorage {
     address: string;
     name: string;
     registeredUsers: User[];
     auctions: Auction[];
+
+    constructor(address: string, marketplaceName: string) {
+        this.address = address ? address : '';
+        this.name = marketplaceName ? marketplaceName : '';
```
Comment (lukaszkinder) [inline src/app/model/marketplace.ts]: both : address and marketplaceName are mandatory - so do not check if they are given - just assign them as they are

File: src/app/admin/service/admin-marketplace-service-mock.service.ts (hunk 2 above)
Comment (KaliszukTomasz) [reply src/app/admin/service/admin-marketplace-service-mock.service.ts]: ok

File: src/app/admin/model/marketplace.ts (hunk 1 above)
Comment (KaliszukTomasz) [reply src/app/admin/model/marketplace.ts]: Done

File: src/app/admin/service/admin-marketplace-service-mock.service.ts (hunk 3 above)
Comment (KaliszukTomasz) [reply src/app/admin/service/admin-marketplace-service-mock.service.ts]: done

**Pull request merged**